domains/grid/components/GridView.vue

```diff
@@ -5,21 +5,39 @@ import { GridItem, GridLayout } from 'grid-layout-plus'
 const GRID_ROW_HEIGHT_PX = 280 // TODO we should calculate this based on grid column width
 const GRID_RESIZE_DEBOUNCE_TIMEOUT_MS = 250
 
-const { isEditingGrid, gridLayout, hasUnsavedGrid, gridColumns } =
-  storeToRefs(useAppStore())
-const address = getCurrentProfileAddress()
+const {
+  isEditingGrid,
+  connectedGridLayout,
+  viewedGridLayout,
+  connectedProfileAddress,
+  hasUnsavedGrid,
+  gridColumns,
+} = storeToRefs(useAppStore())
 const { initializeGridLayout, saveGridLayout, canEditGrid } = useGrid()
-
-const layout = ref<GridWidget[]>([])
 const gridContainer = ref<HTMLElement | null>(null)
+const viewedProfileAddress = getCurrentProfileAddress()
 let resizeTimeout: ReturnType<typeof setTimeout> | null = null
 
+const address = computed(() => getCurrentProfileAddress())
+const layout = ref<GridWidget[]>([])
+
+const currentLayout = computed(() => {
+  if (
+    connectedProfileAddress.value?.toLowerCase() ===
+    viewedProfileAddress.toLowerCase()
+  ) {
+    return connectedGridLayout.value
+  }
+
+  return viewedGridLayout.value
+})
+
 const handleUpdateLayout = (newLayout: GridWidget[]) => {
   if (gridLog.enabled) {
     gridLog('Layout updated', newLayout)
   }
 
-  gridLayout.value = newLayout
+  // connectedGridLayout.value = newLayout
```

Comment:
So `handleUpdateLayout` function only logs now, in some cases. Are there any plans for this specific line or the function itself?

Reply:
Not for now, I'll remove the commented line tho.